### DISCUSSION TITLE
Nonlinear Boundary Integrators?

### DISCUSSION THREAD

**Discussion by j-signorelli** [mfem/mfem#4052] [Q&A] [answered]
Hello,

I have a (hopefully) quick question, I have a nonlinear form like:

$$F(\vec{u})=\int_{\partial\Omega}\lambda(\vec{u})\phi_jd\vec{x}$$
$$\dfrac{\partial F}{\partial\vec{u}}=\int_{\partial\Omega}\dfrac{\partial\lambda}{\partial\vec{u}}\phi_i\phi_jd\vec{x}$$
in context of #4048 -- it is a nonlinear Neumann term for convenience in specifying the boundary condition using a heat flux as I have $\lambda=\dfrac{q}{\rho(\vec{u}) C(\vec{u})}$ for nonlinear heat transfer.

However, I noticed that there is no `AddBoundaryIntegrator` for `NonlinearForm`'s -- why is that?

Thank you very much for any assistance, and happy new year!

**Comment by vladotomov** [comment, marked as answer]
@j-signorelli It's not implemented because probably nobody ever needed it. If you want to add that and make a PR, please go ahead.

There's the `BoundaryFace` option that creates a `FaceElementTransformations` and assembles over the volumetric DOFs, which usually goes with DG spaces (but would also work with CG).

**Reply by j-signorelli** [reply]
@vladotomov Got it, thank you

**Reply by j-signorelli** [reply]
Hi @tzanio, I have requested access to the collaborators MFEM team so I may create a branch off of master to do this, as per contribution guidelines - if it is preferred to do this on my own fork, let me know, thanks!

**Reply by tzanio** [reply]
I added you to the team. Thanks for your help!
